test/functorch/test_control_flow.py

```diff
@@ -1286,7 +1286,6 @@ def fwbw(map_op, f, x, y):
         fake_outs = fwbw(_fake_map, f, x, y)
         self.assertEqual(true_outs, fake_outs)
 
-    # TODO: provide an implementation for all compile modes and re-enable all test
```

Comment:
I didn't read this PR in detail yet. What's the strategy? (can you include the information in the PR body?)

I can brainstorm 2 possible implementations:
1. we add a new lowering for scan that looks like that of while_loop and cond
2. we literally translate the scan into a while_loop

(2) is a cleaner solution, assuming that scan is always host-side and doesn't need to do anything specific in its lowering.

Reply:
Looks like this PR goes with approach 1

Reply:
Yeah, i also think about this. I feel we can decouple the lowering of scan and while_loop.
1. the SequentialScan ir node scan carries clearer information (e.g. the number of iterations) compared with while_loop, where we have to inspect the cond_fn to infer this information. So at least we want to have a dedicated IR node for it. 
2. We could still share code in wrapper codegen between scan and while_loop when necessary.
3. It's not that much work and it allows us to change them separately (e.g. adding new kwargs) without worrying about composability. After all, we only have a few of these so the complexity is managable.

Reply:
I'm good on this subject